scripts/update-projects-version.py

```diff
@@ -0,0 +1,375 @@
+
+    if arguments.force and destination.exists():
+        logging.info(f"Removing destination folder {destination}")
+        shutil.rmtree(destination)
+        destination.mkdir(parents=True)
+
+    if not destination.exists():
+        exit_error(f"Destination folder {destination} does not exist")
+
+    if any(destination.iterdir()):
+        exit_error(f"Destination folder {destination} is not empty")
+
+    logging.info(run_command("operator-sdk version").stdout.strip())
+
+
+def get_version_upstream(file: str) -> semver.VersionInfo:
+    current_version = None
+
+    response: HTTPResponse
+    with request.urlopen(
+        f"https://raw.githubusercontent.com/dora-metrics/pelorus/master/{file}"
+    ) as response:
+        match = re.search(VERSION_PATTERN, response.read().decode())
+        if match:
+            current_version = semver.VersionInfo.parse(match.group())
+    if current_version is None:
+        exit_error(f"Version was not found in {file} in master branch")
+    return current_version
+
+
+def bump_version(
+    version: semver.VersionInfo, arguments: argparse.Namespace
+) -> semver.VersionInfo:
+    # TODO move this to chart-test?
+    if version.build is not None:
+        raise ValueError
+
+    if arguments.major:
+        return version.bump_major()
+    if arguments.minor:
+        return version.bump_minor()
+    if arguments.patch:
+        return version.next_version("patch")
+    if arguments.pre_release:
+        if version.prerelease:
+            return version.bump_prerelease()
+        return version.next_version("patch").bump_prerelease()
+
+
+def get_next_version(file: str, software: str, arguments: argparse.Namespace) -> str:
+    current_version = get_version_upstream(file=file)
+    logging.info(f"Current {software} version (upstream): {current_version}")
+    next_version = bump_version(current_version, arguments=arguments)
+    logging.info(f"Bumping {software} version to {next_version}")
+
+    return str(next_version)
+
+
+def replace_in_file(file: Path, pattern: str, new: str, number_of_changes: int) -> None:
+    current_content = file.read_text()
+    match = re.search(pattern, current_content)
+    if match:
+        file.write_text(current_content.replace(match.group(), new))
+    else:
+        exit_error(f"Pattern {pattern} was not found in {file}")
+    # Verification if the changes were actually applied
+    if file.read_text().count(new) != number_of_changes:
+        exit_error(f"Unexpected changes in {file}")
+
+
+def update_charts_folder(next_chart_version: str) -> None:
+    for file, occurrences in CHARTS_FILES_TO_UPDATE.items():
+        replace_in_file(
+            file=file,
+            pattern=VERSION_PATTERN,
+            new=next_chart_version,
+            number_of_changes=occurrences,
+        )
+
+    run_command(f"helm dep update {PELORUS_CHARTS_FOLDER}")
+    tar_files = (PELORUS_CHARTS_FOLDER / "charts").glob("*.tgz")
```

Comment:
Is this also skipped in .gitignore? Do we really need to remove the tgz if it's not in github?

Reply:
Now that will be automated, I think it is safe, but I would still remove it, to know exactly what we are shipping with the operator